### ISSUE TITLE
Grid Enhancement: use `grid-column-span` instead of `width` for elements that span across multiple grid columns

### ISSUE THREAD

**Issue by soma-m** [terrastruct/d2#1164]
The grid feature is great and D2 is quickly becoming my favorite diagramming language.

Currently this snippet 
```
grid-rows: 2
App 1
App 2
App 3
Linux.width: 400
```
creates this diagram below and conveys all 3 apps are running on Linux.
![image](https://user-images.githubusercontent.com/17534013/230766162-1a0fa99c-23ee-4c55-9f48-bfc9aa9a423d.png)

But since the width of Linux is explicitly specified, the diagram breaks when the app description gets longer.
![image](https://user-images.githubusercontent.com/17534013/230766291-55445eb2-f83b-4876-8ecf-456ce724bf30.png)

**Instead of specifying the exact `width`, Can I use something like `grid-column-span: 3` to get the same effect?**

I feel this is more robust solution because, I often don't care about the exact width and just want to ensure the bottom block covers all 3 columns.

**Comment by alixander** [comment]
> D2 is quickly becoming my favorite diagramming language

Firstly, thank you for the support =) 

Indeed, this is something that was considered. Basically we 
- didn't want to rebuild CSS grid
- minimal keywords

So this is the minimal set to achieve what we want. 

I'm thinking maybe instead of those keywords, we can interpret `width` and `height` as percentages when value is between 0-1 (kind of like https://github.com/terrastruct/d2/pull/1006).

So for that same diagram, you'd just do `linux.width: 1` to say it should be 100%, or `App 1.width: 0.3` to say it should take up 1/3.

**Comment by soma-m** [comment]
I agree. 

Interpreting `width` & `height` as percentages when value is between 0-1 is a very clean solution.

I love it 🙂

**Comment by sparr** [comment]
I disagree. This sort of overloading of a field will lead to problems and edge cases and difficulty for anyone trying to make compatible implementations.

**Comment by alixander** [comment]
The downside I see is that it does force people to do math unnecessarily. Say you have 7 columns and you want you want 3 blocks followed by 1 longer one to fill the rest. you'd have to specify 1/7, 1/7, 1/7, 4/7, which is annoying. will think more about it.
